### PR TITLE
Feature/add bootstrap

### PR DESCRIPTION
**Pull request by UsmanAhmad169**
This branch adds bootstrap, jquery and popper.js to the project. Configuration for these libraries is also done. Partials for a navbar and alerts are also included.

### CODE REVIEW

File: app/assets/stylesheets/application.scss [hunk 1]
```diff
@@ -0,0 +1 @@
+@import "bootstrap/scss/bootstrap";
```
Comment (muhammad-basil) [inline app/assets/stylesheets/application.scss:1]: verify if required here

File: app/javascript/packs/application.js [hunk 2]
```diff
@@ -3,12 +3,13 @@
 // a relevant structure within app/javascript and only use these pack files to reference
 // that code so it'll be compiled.
 
+import 'bootstrap'
```
Comment (muhammad-basil) [inline app/javascript/packs/application.js]: Use require method to make it consistent.

File: app/controllers/pages_controller.rb [hunk 3]
```diff
@@ -0,0 +1,5 @@
+class PagesController < ApplicationController
```
Comment (muhammad-basil) [inline app/controllers/pages_controller.rb]: Add HomeController for this purpose

File: app/views/layouts/_navbar.html.erb [hunk 4]
```diff
@@ -0,0 +1,22 @@
+<nav class="navbar navbar-expand-lg navbar-light bg-light">
+  <a class="navbar-brand" href="#">WorkflowEngine</a>
+  <button class="navbar-toggler" type="button" data-toggle="collapse" data-target="#navbarNav" aria-controls="navbarNav" aria-expanded="false" aria-label="Toggle navigation">
+    <span class="navbar-toggler-icon"></span>
+  </button>
+  <div class="collapse navbar-collapse" id="navbarNav">
```
Comment (muhammad-basil) [inline app/views/layouts/_navbar.html.erb]: follow naming convention for ids: navbar_nav

File: app/controllers/pages_controller.rb (hunk 3 above)
Comment (UsmanAhmad169) [reply app/controllers/pages_controller.rb]: Changed

File: app/javascript/packs/application.js (hunk 2 above)
Comment (UsmanAhmad169) [reply app/javascript/packs/application.js]: Done

File: app/views/layouts/_navbar.html.erb (hunk 4 above)
Comment (UsmanAhmad169) [reply app/views/layouts/_navbar.html.erb]: Done

File: app/assets/stylesheets/application.scss (hunk 1 above)
Comment (UsmanAhmad169) [reply app/assets/stylesheets/application.scss:1]: bootstrap's css does not load without it

File: app/javascript/packs/application.js [hunk 5]
```diff
@@ -3,12 +3,13 @@
...
 // that code so it'll be compiled.
 
+require('bootstrap')
+
 require("@rails/ujs").start()
 require("turbolinks").start()
 require("@rails/activestorage").start()
 require("channels")
 
-
```
Comment (mujeeb-arshad) [inline app/javascript/packs/application.js]: revert

File: app/views/layouts/_alerts.html.erb [hunk 6]
```diff
@@ -0,0 +1,9 @@
+<% flash.each do |_, msg| %>
+	<div class="alert alert-secondary alert-dismissible fade show">
```
Comment (mujeeb-arshad) [inline app/views/layouts/_alerts.html.erb]: remove extra spaces

File: app/views/layouts/_alerts.html.erb [hunk 7]
```diff
@@ -0,0 +1,9 @@
+<% flash.each do |_, msg| %>
```
Comment (mujeeb-arshad) [inline app/views/layouts/_alerts.html.erb:1]: change _ to some better name

File: config/database.yml [hunk 8]
```diff
@@ -13,8 +13,8 @@ default: &default
   adapter: mysql2
   encoding: utf8mb4
   pool: <%= ENV.fetch("RAILS_MAX_THREADS") { 5 } %>
-  username: root
```
Comment (mujeeb-arshad) [inline config/database.yml:16]: add this file to git ignore

File: test/controllers/home_controller_test.rb [hunk 9]
```diff
@@ -0,0 +1,9 @@
+require 'test_helper'
+
```
Comment (mujeeb-arshad) [inline test/controllers/home_controller_test.rb]: delete this test folder completely

File: app/views/layouts/_alerts.html.erb (hunk 7 above)
Comment (UsmanAhmad169) [reply app/views/layouts/_alerts.html.erb:1]: It's not being used anywhere in the block (generally rubocop prefers it this way when something is not being used)

File: test/controllers/home_controller_test.rb (hunk 9 above)
Comment (UsmanAhmad169) [reply test/controllers/home_controller_test.rb]: @mujeeb-arshad , do I need to delete the test folder or home_controller folder

File: app/javascript/packs/application.js (hunk 5 above)
Comment (UsmanAhmad169) [reply app/javascript/packs/application.js]: What to revert?

File: config/database.yml (hunk 8 above)
Comment (UsmanAhmad169) [reply config/database.yml:16]: This file is already in .gitignore but somehow does not get ignored

File: app/views/layouts/_alerts.html.erb (hunk 6 above)
Comment (UsmanAhmad169) [reply app/views/layouts/_alerts.html.erb]: Removed in a new commit

File: app/javascript/packs/application.js (hunk 5 above)
Comment (mujeeb-arshad) [reply app/javascript/packs/application.js]: deletion of extra line

File: test/controllers/home_controller_test.rb (hunk 9 above)
Comment (mujeeb-arshad) [reply test/controllers/home_controller_test.rb]: test folder

File: app/javascript/packs/application.js (hunk 5 above)
Comment (UsmanAhmad169) [reply app/javascript/packs/application.js]: Done

File: test/controllers/home_controller_test.rb (hunk 9 above)
Comment (UsmanAhmad169) [reply test/controllers/home_controller_test.rb]: Done

**Pull request merged**